Lib/test/test_import/__init__.py

```diff
@@ -2413,6 +2416,11 @@ def test_with_reinit_reloaded(self):
 
                 self.assertIs(reloaded.snapshot.cached, reloaded.module)
 
+                if name == f'{self.NAME}_with_state':
+                    loaded.module._clear_module_state()
+                    reloaded.module._clear_module_state()
```

Comment:
This clean-up is only needed for `_testsinglephase_with_state`.
`_testsinglephase_with_reinit` doesn't require this since its state is cleaned right after initialization.